--- a/Makefile	2021-07-23 07:21:36.000000000 -0700
+++ b/Makefile	2023-04-05 13:21:23.033192841 -0700
@@ -1,4 +1,4 @@
-GCLDIR := $(if $(GCLDIR),$(GCLDIR),../gclib)
+GCLDIR := $(if $(GCLDIR),$(GCLDIR),gclib)
 
 SEARCHDIRS := -I. -I${GCLDIR}
 
@@ -79,10 +79,10 @@
  
 .PHONY : all
 
-all static release debug memcheck memdebug profile gprof prof: ../gclib gffread
+all static release debug memcheck memdebug profile gprof prof: gclib gffread
 
-../gclib:
-	git clone https://github.com/gpertea/gclib.git ../gclib
+./gclib:
+	git clone https://github.com/gpertea/gclib.git gclib
 
 $(OBJS) : $(GCLDIR)/GBase.h $(GCLDIR)/gff.h
 gffread.o : gff_utils.h $(GCLDIR)/GBase.h $(GCLDIR)/gff.h
